internal/versions/versions_test.go

```diff
@@ -650,9 +653,556 @@ var _ = Describe("GetReleaseImageByURL", func() {
 	})
 })
 
+var _ = Describe("SyncReleaseImages", func() {
+	var (
+		db          *gorm.DB
+		dbName      string
+		ctrl        *gomock.Controller
+		err         error
+		handler     *handler
+		mockRelease *oc.MockRelease
+	)
+
+	type OpenShiftReleaseImageNoGorm struct {
+		OpenshiftVersion string
+		Channel          string
+		CpuArchitecture  string
+		Version          string
+		Url              string
+	}
+
+	defaultReleaseSources := models.ReleaseSources{
+		{
+			OpenshiftVersion: swag.String("4.13"),
+			UpgradeChannels: []*models.UpgradeChannel{
+				{
+					CPUArchitecture: swag.String("x86_64"),
+					Channels:        []string{"stable"},
+				},
+				{
+					CPUArchitecture: swag.String("s390x"),
+					Channels:        []string{"stable"},
+				},
+			},
+		},
+		{
+			OpenshiftVersion: swag.String("4.14"),
+			UpgradeChannels: []*models.UpgradeChannel{
+				{
+					CPUArchitecture: swag.String("arm64"),
+					Channels:        []string{"stable", "candidate"},
+				},
+				{
+					CPUArchitecture: swag.String("ppc64le"),
+					Channels:        []string{"fast", "eus"},
+				},
+			},
+		},
+	}
+
+	BeforeEach(func() {
+		db, dbName = common.PrepareTestDB()
+		ctrl = gomock.NewController(GinkgoT())
+		httpmock.Activate()
+		handler, err = NewHandler(common.GetTestLog(), mockRelease, defaultReleaseImages, defaultReleaseSources, nil, "", nil, db)
+	})
+
+	AfterEach(func() {
+		ctrl.Finish()
+		common.DeleteTestDB(db, dbName)
+		httpmock.DeactivateAndReset()
+	})
+
+	Context("existAlready", func() {
+		It("Should be True when exist", func() {
+			ocpReleaseImage := common.OpenShiftReleaseImage{
+				OpenshiftVersion: "4.14.1",
+				Channel:          "stable",
+				CpuArchitecture:  "x86_64",
+				Url:              fmt.Sprintf(OcpImageUrlTemplate, "stable", "4.14.1"),
+			}
+
+			db.Create(&ocpReleaseImage)
+			result := existAlready(db, ocpReleaseImage)
```

Comment:
It's good practice to test only public methods, so that we don't have tests that are too fragile. We still most likely want to cover cases where the image already exists, testing it implicitly.